### ISSUE TITLE
Version Android Manifest inserts an empty string instead of zero for version code

### ISSUE THREAD

**Issue by deadlydog** [rfennell/vNextBuild#387]
### Azure DevOps Extension you are using
Version Android Manifest

### Where are you running it?
Azure DevOps Service (VSTS)

### Version of Extension/Task
2.0.4

### Expected behaviour and actual behaviour
When using a version number like 1.0.0 I expect it to use "1.0" in the Version Name Pattern and "0" in the Version Code Pattern. However, when it attempts to build the project I get the error message:

> src\JokeGenerator\JokeGenerator.Android\obj\Release\MonoAndroid81\android\manifest\AndroidManifest.xml(3,0): Error APT0000: String types not allowed (at 'versionCode' with value '').

These are the Task settings I am using:
![image](https://user-images.githubusercontent.com/1187140/46934956-cd7a4380-d016-11e8-9cae-cd040a6bb715.png)
And the build number format is `$(date:yyyyMMdd)-1.0.0.$(rev:r)`.

If I simply change the build number format to `$(date:yyyyMMdd)-1.0.1.$(rev:r)` then everything works properly.

### Steps to reproduce the problem

Use a version number that has a zero for the 3rd part and attempt to build the Android csproj. It will fail because the version code is blank instead of zero.

### Additional Info

Here is the output from the Version Android Manifest task:

```
2018-10-15T05:56:46.9481266Z ##[section]Starting: Version Android Manifest
2018-10-15T05:56:46.9485245Z ==============================================================================
2018-10-15T05:56:46.9485323Z Task         : Version Android Manifest
2018-10-15T05:56:46.9485386Z Description  : Applies a version to an Android Manifest via the versionCode property based on the build number.
2018-10-15T05:56:46.9485433Z Version      : 2.0.4
2018-10-15T05:56:46.9485490Z Author       : Black Marble
2018-10-15T05:56:46.9485536Z Help         : Version: 2.0.4. [More Information](https://github.com/rfennell/vNextBuild/wiki/Version-Assemblies-and-Packages-Tasks/)
2018-10-15T05:56:46.9485585Z ==============================================================================
2018-10-15T05:56:47.2156898Z Source Directory:  D:\a\1\s
2018-10-15T05:56:47.2157418Z Filename Pattern: AndroidManifest.xml
2018-10-15T05:56:47.2157730Z Version Number/Build Number: 20181014-1.0.0.1
2018-10-15T05:56:47.2157867Z Version Filter to extract build number: \d+\.\d+\.\d+
2018-10-15T05:56:47.2158545Z Version Number Format: {1}.{2}
2018-10-15T05:56:47.2158700Z Version Code Format: {3}
2018-10-15T05:56:47.2158821Z Output: Version Number Parameter Name: OutputedVersion
2018-10-15T05:56:47.2366236Z Found 1 files to update.
2018-10-15T05:56:47.2366568Z Extracted Version: 1.0.0
2018-10-15T05:56:47.2383852Z Version Name will be: 1.0
2018-10-15T05:56:47.2385331Z Version Code will be 0
2018-10-15T05:56:47.2385431Z Will apply versionName: 1.0 and versionCode: 0 to 1 files.
2018-10-15T05:56:47.2392954Z Set the output variable 'OutputedVersion' with the value 1.0.0
2018-10-15T05:56:47.2456458Z ##[section]Finishing: Version Android Manifest
```

It says:

> Version Code will be 0

But it seems that it uses an empty string. Unless zero is simply not a valid version code?

Hopefully you can address this one, even if the problem is that zero is not a valid version code, then just increment it to 1.

Thanks for the great extensions!

**Comment by ChrisLGardner** [comment]
After discussion with @rfennell we're going to make it so the task fails if the Version Code would be 0. If we auto increment it to 1 then it'll almost certainly cause problems further down the line, mainly when you increment the minor version number.

I'd suggest making your Version Code {2}{3} or {1}{2}{3} to include the other version numbers into it and increase the chance it is unique.

**Comment by ChrisLGardner** [comment]
Released in 2.1.x

**Comment by deadlydog** [comment]
Awesome, thanks for the info Chris. I've never created an Android app before so I didn't realize those specifics about the Version Code. I could potentially see the value decreasing accidentally when using `{1}{2}{3}` (e.g. going from version 1.10.3 to version 2.0.1), so I've opted to instead use `$(Build.BuildId)` as that should be an ever increasing integer value. We'll see how that goes. Thanks!
